app/models/concerns/hostext/taxonomies.rb

```diff
@@ -0,0 +1,20 @@
+module Hostext
+  module Taxonomies
+    extend ActiveSupport::Concern
+
+    included do
+      validates :organization_id, :presence => true, :if => ->(host) { host.managed? && SETTINGS[:organizations_enabled] }
+      validates :location_id,     :presence => true, :if => ->(host) { host.managed? && SETTINGS[:locations_enabled] }
```

Comment:
nitpick: I would extract the setting check condition to the entire line vs a if within the lambda.